modules/slugs/admin/slugs-migrate-qts.php

```diff
@@ -0,0 +1,153 @@
+        if ( $results ) {
+            $msg[] = sprintf( __( "Found %s slugs from $table.", 'qtranslate' ), $results );
+        }
+    };
+
+    $msg = [];
+    $count_slugs( $wpdb->postmeta, QTX_SLUGS_META_PREFIX, $msg );
+    $count_slugs( $wpdb->termmeta, QTX_SLUGS_META_PREFIX, $msg );
+    if ( ! empty( $msg ) ) {
+        // Found some post/term meta with the new keys, no migrate to suggest (it can still be done manually).
+        return '';
+    }
+
+    $msg = [];
+    $count_slugs( $wpdb->postmeta, QTX_SLUGS_LEGACY_QTS_META_PREFIX, $msg );
+    $count_slugs( $wpdb->termmeta, QTX_SLUGS_LEGACY_QTS_META_PREFIX, $msg );
+
+    return empty ( $msg ) ? $msg : implode( '<br>', $msg );
+}
+
+/**
+ * Migrate slugs meta by migrating the legacy QTS postmeta and termmeta to QTX.
+ * Attention: current slugs meta are deleted.
+ *
+ * @param bool $db_commit true to commit changes, false for dry-run mode.
+ *
+ * @return string messages giving details.
+ */
+function qtranxf_slugs_migrate_qts_meta( $db_commit ) {
+    global $wpdb;
+
+    $new_prefix = QTX_SLUGS_META_PREFIX;
+    $old_prefix = QTX_SLUGS_LEGACY_QTS_META_PREFIX;
+
+    /**
+     * Generic function that migrates QTS meta to QTX meta.
+     *
+     * @param string $table name of the meta table (postmeta, termmeta)
+     * @param string[] $msg array of messages, updated
+     *
+     * @return void
+     */
+    $migrate_meta = function ( $table, &$msg ) use ( $wpdb, $old_prefix, $new_prefix ) {
+        $results = $wpdb->query( "DELETE FROM $table WHERE meta_key like '$new_prefix%'" );
+        $msg[]   = sprintf( __( "Deleted %s rows from $table (%s).", 'qtranslate' ), $results ?: '0', $new_prefix );
+        // Rename meta keys.
+        $results = $wpdb->query( "UPDATE $table SET meta_key = REPLACE(meta_key, '$old_prefix', '$new_prefix') WHERE meta_key LIKE '$old_prefix%'" );
+        $msg[]   = sprintf( __( "Migrated %s rows from $table (%s).", 'qtranslate' ), $results ?: '0', $old_prefix );
+    };
+
+    $msg = [];
+    $wpdb->query( "START TRANSACTION" );
+    $migrate_meta( $wpdb->postmeta, $msg );
+    $migrate_meta( $wpdb->termmeta, $msg );
+    if ( $db_commit ) {
+        $wpdb->query( "COMMIT" );
+    } else {
+        $wpdb->query( "ROLLBACK" );
+    }
+
+    return implode( '<br>', $msg );
+}
+
+/**
+ * Migrate legacy QTS options to QTX.
+ * Attention: current slugs options are deleted.
+ *
+ * @param bool $db_commit true to commit changes, false for dry-run mode.
+ *
+ * @return string messages giving details.
+ */
+function qtranxf_slugs_migrate_qts_options( $db_commit ) {
+    $msg = [];
+
+    $old_options = get_option( QTX_OPTIONS_MODULE_SLUGS );
+    if ( $old_options ) {
+        if ( $db_commit ) {
+            delete_option( QTX_OPTIONS_MODULE_SLUGS );
+        }
+        $msg[] = sprintf( __( "Deleted %s types from options.", 'qtranslate' ), count( $old_options ) );
+    }
```

Comment:
I think this part should be moved inside the `if ( $qts_options ) {`, no reason to remove anything if no old option is there to be migrated.